File: 07/main.cpp
#include <iostream>
#include <vector>

using namespace std;
class Solution {
public:
	int Fibonacci(int n) {
		if (n < 0)
			return NULL;
		else if (n == 0 || n == 1)
			return n;
		
		vector<int> f;
		f.push_back(0);
		f.push_back(1);
		for (int i = 2; i <= n; i++)
			f.push_back(f[i - 1] + f[i - 2]);
		return f[n];
	}
};

int main(){
	Solution so;
	int n;
	cin >> n;
	int f = so.Fibonacci(n);
	cout << "Item n of the Fibonacci sequence:" << f << endl;

	getchar(); getchar();
	return 0;
}

File: 07/main.py
# -*- coding:utf-8 -*-
class Solution:
    def __init__(self):
        pass
    def Fibonacci(self, n):
        if n<0:
            return None
        if n==0 or n==1:
            return n
        
        f=[]
        f.append(0)
        f.append(1)
        for i in range(2,n+1):
            f.append(f[i-1]+f[i-2])
        return f[n]
    
so=Solution()
x=int(raw_input())
f=so.Fibonacci(x)
print "Item n of the Fibonacci sequence:{}".format(f)